File: Day_06/day6.rb
# Day 6 Part 1
p IO.read("data.txt").split(?\n*2).sum{|x|x.chars.uniq.sort.join.strip.size}
# Day 6 Part 2
p IO.read("data.txt").split(?\n*2).sum{|g|g.split.map(&:chars).inject(:&).size}

File: 2020/Day_06/day6.py
# Part 1
print("Part 1: {}".format(sum(len(set(i.replace('\n', ''))) for i in open('data.txt').read().split('\n\n'))))
# Part 2
print("Part 2: {}".format(sum(len(set.intersection(*[set(t) for t in i.split("\n")])) for i in open('data.txt').read().split("\n\n"))))
